ambry-rest/src/test/java/com.github.ambry.rest/NettyMultipartRequestTest.java

```diff
@@ -347,6 +350,49 @@ public void prepareTest() throws Exception {
     }
   }
 
+  /**
+   * Tests to make sure the max allowed size for multipart requests is enforced.
+   * @throws Exception
+   */
+  @Test
+  public void sizeLimitationTest() throws Exception {
+    int blobPartSize = 1024;
+    // encoding 1024 bytes of data results in 1209 readable bytes from the encoder.
+    int encodedSize = 1209;
+    long[] maxSizesAllowed = {encodedSize + 1, encodedSize, encodedSize - 1, 0};
+    for (long maxSizeAllowed : maxSizesAllowed) {
+      InMemoryFile[] files = {new InMemoryFile(RestUtils.MultipartPost.BLOB_PART,
+          ByteBuffer.wrap(TestUtils.getRandomBytes(blobPartSize)))};
+      HttpRequest httpRequest = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, "/");
+      HttpPostRequestEncoder encoder = createEncoder(httpRequest, files);
+      NettyMultipartRequest request =
+          new NettyMultipartRequest(encoder.finalizeRequest(), new MockChannel(), NETTY_METRICS, Collections.emptySet(),
+              maxSizeAllowed);
+      assertTrue("Request channel is not open", request.isOpen());
+      long currentSizeAdded = 0;
+      while (!encoder.isEndOfInput()) {
+        // Sending null for ctx because the encoder is OK with that.
+        HttpContent httpContent = encoder.readChunk(PooledByteBufAllocator.DEFAULT);
+        int readableBytes = httpContent.content().readableBytes();
+        if (currentSizeAdded + readableBytes <= maxSizeAllowed) {
+          request.addContent(httpContent);
+        } else {
+          assertTrue("Max size [" + maxSizeAllowed + "] must be lesser than content size: " + encodedSize,
+              maxSizeAllowed < encodedSize);
+          try {
+            request.addContent(httpContent);
```

Comment:
how do you ensure that this block is being execute for those interested ones?

Reply:
Added a check to make sure that it fails (or doesn't) depending on the relation b/w `encodedSize` and `maxSizeAllowed`

Reply:
looks good